docs/docs-beta/docs/guides/io-managers.md

```diff
@@ -4,43 +4,85 @@ sidebar_position: 50
 sidebar_label: "I/O managers"
 ---
 
-I/O managers in Dagster provide a way to separate the code that's responsible for logical data transformation from the code that's responsible for reading and writing the results.  This can help reduce boiler plate code and make it easy to swap out where your data is stored.
+I/O managers in Dagster allow you to keep the code for data processing separate from the code for reading and writing data. This reduces repetitive code and makes it easier to change where your data is stored.
+
+In many Dagster pipelines, assets can be broken down as the following steps:
+
+1. Reading data a some data store into memory
+2. Applying in-memory transform
+3. Writing the transformed data to a data store
+
+For assets that follow this pattern, an I/O manager can streamline the code that handles reading and writing data to and from a source.
 
 <details>
 <summary>Prerequisites</summary>
 
-- Familiarity with [Assets](/concepts/assets)
-- Familiarity with [Resources](/concepts/resources)
+To follow the steps in this guide, you'll need familiarity with:
+
+- [Assets](/concepts/assets)
+- [Resources](/concepts/resources)
 </details>
 
-## Extract I/O logic from an asset into a reusable I/O manager
+## Before you begin
```

Comment:
This section is based off of our existing docs ([link](https://docs.dagster.io/concepts/io-management/io-managers#when-to-use-io-managers)), which I tried to simplify. It may need more work, though - feedback welcome.